change a few global Lua variables to be local

quietly-turning opened the pull request:
Running StepMania with the command line argument of `--ExportLuaInformation` while the current 5.1's default theme (Lambda) was selected added some seemingly stray Lua variables to the Constants table in ./Docs/Luadoc/Lua.xml.

I spotted them here: http://dguzek.github.io/Lua-For-SM5/LuaAPI#Constants

I'm assuming `num`, `bPreference`, and `bCategory` don't need global scope, so I added the `local` keyword where appropriate.

shakesoda commented on Themes/_fallback/Scripts/03 CustomSpeedMods.lua:
```diff
@@ -70,6 +70,8 @@ end
 -- Parses a speed mod and returns the pair (type, number) or nil if parsing
 -- failed.
 local function CanonicalizeMod(mod)
+	local num = nil
```
this should be dropped in favor of just local num = tonumber(...) on next line

quietly-turning replied:
Yeah, good call.  Updated and force-pushed.

shakesoda reviewed (changes requested):
seems to me fine, i would just change that tonumber line

quietly-turning committed:
change a few global Lua variables to be local

Running StepMania with the command line argument of --ExportLuaInformation while the current 5.1's default theme (Lambda) was selected added some seemingly stray Lua variables to the Constants table in ./Docs/Luadoc/Lua.xml.

I spotted them here: http://dguzek.github.io/Lua-For-SM5/LuaAPI#Constants

I'm assuming num, bPreference, and bCategory don't need global scope, so I added the local keyword where appropriate.
